### PR TITLE
Add SBU Captions Photo Dataset

### PR DESCRIPTION
**Pull request by thomasw21**

### CODE REVIEW

**Review by mariosasko — APPROVED** [review]
Very clean implementation. Great job, @thomasw21! 

P.S. I added minor improvements to the script (support for streaming) and the card (faster downloading).

File: datasets/sbu_captions/sbu_captions.py [hunk 1]
```diff
@@ -76,24 +76,26 @@ def _info(self):
...
-    def _generate_examples(self, annotations_file):
-        # Download all images
-        with open(annotations_file, "r", encoding="utf-8") as fi:
-            annotations = json.load(fi)
+    def _generate_examples(self, files):
+        annotations = None
+        for path, f in files:
+            if path.endswith("sbu-captions-all.json"):
+                annotations = json.loads(f.read().decode("utf-8"))
+                break
```
Comment (thomasw21) [inline datasets/sbu_captions/sbu_captions.py:95]: Why is this a better way of doing it?

File: datasets/sbu_captions/README.md [hunk 2]
```diff
@@ -53,58 +53,66 @@ pretty_name: SBU Captioned Photo Dataset
 
 - **Homepage:** http://www.cs.virginia.edu/~vicente/sbucaptions/
 - **Repository:**
-- **Paper:** http://tamaraberg.com/papers/generation_nips2011.pdf
+- **Paper:** [Im2Text: Describing Images Using 1 Million Captioned Photographs]https://papers.nips.cc/paper/2011/hash/5dd9db5e033da9c6fb5ba83c7a7ebea9-Abstract.html
```
[the remark this answers is not in the document]
Comment (mariosasko) [reply datasets/sbu_captions/README.md]: Yes, but it's fixed in the final diff.

File: datasets/sbu_captions/sbu_captions.py (hunk 1 above)
Comment (thomasw21) [reply datasets/sbu_captions/sbu_captions.py:95]: Got it! Thanks for clarification.

**Review by lhoestq — APPROVED** [review]
thanks !

**Pull request merged**